The gapis server has exited with an error code of: 2

GAPID Version: 1.5.0:developer
OS: linux amd64

Reproduction steps:
- tracing the graphicsfuzz vulkan worker, asking for **10 frames** (the app produces only 5 frames)
- open capture

Maybe this appears when we request more frames than the app produces?

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf74ed2]

goroutine 80 [running]:
github.com/google/gapid/core/app/crash.Crash(0x5f65500, 0xbfc65f0)
	core/app/crash/crash.go:89 +0xe1
github.com/google/gapid/core/app/crash.handler()
	core/app/crash/crash.go:56 +0x54
panic(0x5f65500, 0xbfc65f0)
	GOROOT/src/runtime/panic.go:513 +0x1b9
github.com/google/gapid/gapir/client.(*Connection).Shutdown(0x0, 0x71a1180, 0xc005f46f00, 0x0, 0x0)
	gapir/client/connection.go:117 +0x42
github.com/google/gapid/gapir/client.(*session).close(0xc000613e50, 0x71a1180, 0xc005f46f00)
	gapir/client/session.go:377 +0x4c
github.com/google/gapid/gapir/client.(*session).init(0xc000613e50, 0x71a1180, 0xc005f46f00, 0x71e83a0, 0xc0000efa10, 0xc000100200, 0x0, 0x0, 0x0, 0x0, ...)
	gapir/client/session.go:80 +0x1c3
github.com/google/gapid/gapir/client.(*Client).Connect(0xc0000ef670, 0x71a1180, 0xc005f46f00, 0x71e83a0, 0xc0000efa10, 0xc000100200, 0x0, 0x0, 0x0)
	gapir/client/client.go:66 +0x37f
github.com/google/gapid/gapis/replay.(*manager).connect(0xc00000df80, 0x71a1180, 0xc005f46e40, 0x71e83a0, 0xc0000efa10, 0xc000100200, 0x0, 0x0, 0x0)
	gapis/replay/replay_connection.go:180 +0x24e
github.com/google/gapid/gapis/replay.(*manager).execute(0xc00000df80, 0x71a1180, 0xc005f46e40, 0x71e83a0, 0xc0000efa10, 0xe17bebb8b73d1771, 0xa8fdcc7bca999dc5, 0xec8c272f17de7e05, 0x5b00f0bb3c6c82e6, 0x8cd498d92e1b7b5c, ...)
	gapis/replay/batch.go:270 +0xa66
github.com/google/gapid/gapis/replay.(*manager).batch.func1(0x71e83a0, 0xc0000efa10, 0xc0002f8290, 0x3c6c82e6ec8c272f, 0x2e1b7b5c5b00f0bb, 0xb73d17718cd498d9, 0xca999dc5e17bebb8, 0x17de7e05a8fdcc7b, 0x6517220, 0xc005eafce0, ...)
	gapis/replay/batch.go:86 +0x551
github.com/google/gapid/gapis/replay.(*manager).batch(0xc00000df80, 0x71a1180, 0xc005f4c180, 0xc0004d2820, 0x5, 0x5, 0x6495020, 0x70ac870, 0x64c2100, 0xc00012ef00, ...)
	gapis/replay/batch.go:87 +0x50c
github.com/google/gapid/gapis/replay.(*manager).batch-fm(0x71a1180, 0xc00023e0c0, 0xc0004d2820, 0x5, 0x5, 0x6495020, 0x70ac870, 0x64c2100, 0xc00012ef00, 0x1)
	gapis/replay/manager.go:146 +0x95
github.com/google/gapid/gapis/replay/scheduler.(*bin).exec(0xc005bf6380, 0x71a1180, 0xc00023e0c0, 0xc000390e20)
	gapis/replay/scheduler/scheduler.go:247 +0x2d3
github.com/google/gapid/gapis/replay/scheduler.(*Scheduler).run(0xc000306e80, 0x71a1180, 0xc00023e0c0)
	gapis/replay/scheduler/scheduler.go:180 +0x958
github.com/google/gapid/gapis/replay/scheduler.New.func1()
	gapis/replay/scheduler/scheduler.go:74 +0x50
github.com/google/gapid/core/app/crash.Go.func1(0xc000306ea0)
	core/app/crash/crash.go:65 +0x45
created by github.com/google/gapid/core/app/crash.Go
	core/app/crash/crash.go:63 +0x3f
```

--- a/gapir/client/session.go
+++ b/gapir/client/session.go
@@ -374,7 +374,9 @@ func (s *session) onClose(f func()) {
 }
 
 func (s *session) close(ctx context.Context) {
-	s.conn.Shutdown(ctx)
+	if s.conn != nil {
+		s.conn.Shutdown(ctx)
+	}
 	for _, f := range s.closeCBs {
 		f()
 	}

